.codesandbox/sandbox/src/end-to-end-test.ts

```diff
@@ -0,0 +1,42 @@
+import * as SolidClient from "@inrupt/solid-client";
+import {
+  Session,
+  getClientAuthenticationWithDependencies,
+} from "@inrupt/solid-client-authn-browser";
+import { getHelpers } from "./end-to-end-test-helpers";
+
+const session = new Session({
+  clientAuthentication: getClientAuthenticationWithDependencies({}),
+});
+
+const loginButton = document.getElementsByTagName("button")[0];
+
+loginButton.addEventListener("click", async (e) => {
+  const idp = (document.getElementById("idp") as HTMLInputElement).value;
+  await session.login({
+    oidcIssuer: idp,
+    redirectUrl: document.location.href,
+  });
+});
+
+if (!session.info.isLoggedIn) {
+  session.handleIncomingRedirect(window.location.href).then(async () => {
```

Comment:
I think this is alway going to be true, since the session cannot be logged in before `handleincomingRedirect` has been called. In any case, if the session is already logged in, calling `handleincomingRedirect` just returns the current session information, so I think the outer `if` can be removed.

Reply:
Hmm that's great... I copied that from here though: https://docs.inrupt.com/developer-tools/javascript/client-libraries/tutorial/authenticate/#example

Reply:
Ah, I presume that is a remnant from when sessions could survive between refreshes, in which case the outer `if` was useful?